### PR TITLE
Increase the effectiveness of bomb armor

### PR DESCRIPTION
**Pull request by datlo**
Bomb resistance is currently a little underwhelming. 

You get a chance to prevent gibbing at max explosion strength, a chance to divide damage by 1.5 at medium strength, and a chance to divide by 2 at low strength.

While reducing the damage is nice, it's not usually the raw damage that makes explosion so deadly, it's the knock out combined with breach, aswell as the limb loss.

This PR buffs bomb resistance in the following ways : 

1. Having any amount of bomb resistance will make explosions stun you for 10 instead of knocking you out for 10.
2. This stun is reduced in time by the amount of bomb resistance you have, divided by 10. This mean with a bomb resistance of 100 (bomb suit and hood) you will not be stunned.
3. You now have a chance to negate limb loss based on your bomb resistance. With a bomb resistance of 100, you will never lose a limb from explosions.

For reference here are the bomb armor values of most relevant armors : 

Standard hardsuits : 10
Wizard robes : 20
Captain/HOS/sec armor : 25
Syndi hardsuit : 35
CE/mining hardsuits : 50
Syndi elite hardsuit : 55
Bomb suit : 100

These changes would make these values more relevant, and offer more counterplay to explosives.

:cl:
balance: Armors with bomb resistance are more effective overall at explosive protection.
/:cl:

### CODE REVIEW

**Comment by TDSSS** [comment]
Overall, I think this is a great change to make a mostly unused armor value useful, but I don't think I agree with
>Having any amount of bomb resistance will make explosions stun you for 10 instead of knocking you out for 10.
This kind of threshold-based armor feels weird, where a single point of bomb resistance does something.

**Comment by datlo** [comment]
I didnt want to change how bombs affected an unarmored target, since the point is to make armor more relevant. I think it's important that even a standard security armor provides protection worth going for.

Perhaps I could change it to have it check for armor on your chest instead of overall for that condition, so that you arent knockout immune with just a sec helmet or other minor protection items? I will think about this.

**Comment by lolzyking** [comment]
Dalto it actualy makes more sense to only check for head slot bomb resistance, not chest, because brains.

**Comment by Spacemanspark** [comment]
From a realism perspective maybe, but from a gameplay perspective it would be better to check for the chest.

**Comment by datlo** [comment]
Added weakening on top of stunning so that bombs knock you on the floor still, instead of just staying up and stunned.

I've decided to keep the PR as is otherwise, adding a check for specifically chest armor isnt needed and wont matter in nearly all situations.

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -281,17 +287,22 @@
 				while(limbs_affected != 0 && valid_limbs.len > 0)
 					processing_dismember = pick(valid_limbs)
 					if(processing_dismember.limb_name != "chest" && processing_dismember.limb_name != "head" && processing_dismember.limb_name != "groin")
-						processing_dismember.droplimb(1,DROPLIMB_SHARP,0,1)
+						if(!prob(bomb_armor))
+							processing_dismember.droplimb(1, DROPLIMB_SHARP, 0, 1)
```
[the remark this answers is not in the document]
Comment (AffectedArc07) [reply code/modules/mob/living/carbon/human/human.dm]: Note to self: this proc uses Booleans not a numerical property value

**Comment by Fox-McCloud** [comment]
Need to look at a few things before I come to a conclusion on how good or bad this is, design wise.

**Comment by datlo** [comment]
Fixed merge conflict

**Comment by datlo** [comment]
TG buffed bomb armor in their own way, I'll look into it and open a new PR later with those (or some of those) changes.

**Pull request closed without merging**